packages/invoices/src/InboundShipment/DetailView/api.ts

```diff
@@ -201,31 +127,31 @@ export const getInboundShipmentDetailViewApi = (
   onRead: async (id: string): Promise<Invoice> => {
     const result = await api.invoice({ id, storeId });
 
-    const invoice = invoiceGuard(result);
-    const lineNodes = linesGuard(invoice.lines);
-    const lines: InvoiceLine[] = lineNodes.map(line => {
-      const stockLine = line.stockLine
-        ? stockLineGuard(line.stockLine)
-        : undefined;
-      const location = line.location ? locationGuard(line.location) : undefined;
-      const item = line.item ? itemGuard(line.item) : undefined;
+    const invoice = result.invoice;
+
+    if (invoice.__typename === 'InvoiceNode') {
+      const lineNodes = invoice.lines.nodes;
+      const lines: InvoiceLine[] = lineNodes.map(line => {
+        const stockLine = line.stockLine;
+        const location = line.location;
+
+        return {
+          ...line,
+          stockLine,
+          location,
+          stockLineId: stockLine?.id ?? '',
+          invoiceId: invoice.id,
+          unitName: line.item?.unitName ?? '',
+        };
+      });
```

Comment:
this isn't ideal and it would be good to revisit and pay a little closer attention, probably look at removing the `InvoiceLine` in favour of a fragment